### PR TITLE
update setuptools version

### PR DESCRIPTION
**Pull request by zpcore**
# Description

Fix nightly test issue with setuptools version.

This is the issue that setuptools version >= 71.X doesn't work: https://github.com/pypa/setuptools/issues/4483

# Tests

https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/pytorchxla-torchbench-release/grid?dag_run_id=manual__2024-07-23T17%3A01%3A51.682195%2B00%3A00&task_id=torchbench_all-v4-8.provision.setup&tab=logs

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by vanbasten23** [comment]
Could you summarize the reason why we need to select the setuptools version ">=70.0.0,<71.0.0" in the PR description?

Also, do you need to add a TODO to remove the version constraint later?

**Review by vanbasten23 — APPROVED** [review]

**Comment by zpcore** [comment]
> Could you summarize the reason why we need to select the setuptools version ">=70.0.0,<71.0.0" in the PR description?
> 
> Also, do you need to add a TODO to remove the version constraint later?

Added in the description. Personally I have no plan to remove the version constraint in the near term. This requires other packages to update their setup.py accordingly, which may take time.
